objects/f42a3e/contained/a15998/script.lua

```diff
@@ -0,0 +1,418 @@
+--local rescan
+local currentBoard
+local boardTypes = {"Balanced", "Thematic"}
+
+function onLoad()
+    self.createButton({
+        click_function = "nullFunc",
+        label = "Place Island Board Here",
+        font_color = {1,1,1},
+        position = {0,0.1,0},
+        font_size = 300,
+        width = 0,
+        height = 0,
+    })
+    Color.Add("SoftBlue", Color.new(0.53,0.92,1))
+    Color.Add("SoftYellow", Color.new(1,0.8,0.5))
+    Wait.time(scan, 0.5, -1)
+    --rescan = false
+end
+
+function scan()
+    local objs = upCast(self, 1, 0, {"Tile"})
+    if #objs ~= 1 then
+	currentBoard = nil
+        clearButtons()
+        return
+    end
+    currentBoard = objs[1]
+    local type
+    for _,tag in pairs(boardTypes) do
+        if currentBoard.hasTag(tag) then
+            type = tag
+            break
+        end
+    end
+    createButtons(currentBoard, type)
+end
+
+function clearButtons()
+    local buttons = self.getButtons()
+    for i=2,#buttons do
+        self.removeButton(buttons[i].index)
+    end
+end
+
+function createButtons(board, type)
+    clearButtons()
+    local tooltip = "[b]Balanced[/b] boards will additionally setup a Beasts in the lowest-numbered land without other setup and a Disease in land #2.\n[b]Thematic[/b] boards will not."
+    self.createButton({
+        click_function = "nullFunc",
+        label          = "Type",
+        position       = {3,0.1,16},
+        width          = 830,
+        height         = 330,
+        font_size      = 300,
+	tooltip	       = tooltip,
```

Comment:
These buttons are all next to each other. I gave the uninteractable ones the tooltip as well so that 1. players don't miss it and 2. it will match the intuition to hover over words / larger buttons.